x/mongo/driver/topology/topology_test.go

```diff
@@ -693,6 +693,217 @@ func TestTopologyConstruction(t *testing.T) {
 	})
 }
 
+type mockLogSink struct {
+	msgs []string
+}
+
+func (s *mockLogSink) Info(_ int, msg string, _ ...interface{}) {
+	s.msgs = append(s.msgs, msg)
+}
+func (*mockLogSink) Error(error, string, ...interface{}) {
+	// Do nothing.
+}
+
+// Note: SRV connection strings are intentionally untested, since initial
+// lookup responses cannot be easily mocked.
+func TestTopologyConstructionLogging(t *testing.T) {
+	sink := &mockLogSink{}
+	loggerOptions := options.
```

Comment:
Optional: if these are re-defined for each subtest, then we can parallelize the subtests.